app/Http/Controllers/API/v1/AuthController.php

```diff
@@ -0,0 +1,122 @@
+use Illuminate\Support\Facades\Validator;
+use Psy\Util\Json;
+
+class AuthController extends ResponseController
+{
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     * @api v1
+     * @todo refactor this
+     */
+    public function register(Request $request):JsonResponse {
+        $validator = Validator::make($request->all(), [
+            'username' => 'required|string|unique:users',
+            'name'     => 'required|string|',
+            'email'    => 'required|string|email|unique:users',
+            'password' => 'required|min:3|confirmed',
+        ]);
+        if ($validator->fails()) {
+            return response()->json([
+                                        'status' => 'error',
+                                        'errors' => $validator->errors()
+                                    ], 422);
+        }
+        $input             = $request->all();
+        $input['password'] = Hash::make($input['password']);
+        $user              = User::create($input);
+
+        if ($user) {
+            $userToken = $user->createToken('token');
+            return $this->sendResponse([
+                                           'token'      => $userToken->accessToken,
+                                           'message'    => 'Registration successful.',
+                                           'expires_at' => $userToken->token->expires_at->toIso8601String()
+                                       ]);
+        }
+        $error = "Sorry! Registration is not successful.";
+        return $this->sendError($error, 401);
+    }
+
+    /**
+     * Login
+     * @param Request $request
+     * @return JsonResponse
+     * @api v1
+     */
+    public function login(Request $request): JsonResponse {
+        $validator = Validator::make($request->all(), [
+            'email'    => 'required|string|email',
+            'password' => 'required'
+        ]);
+
+        if ($validator->fails()) {
+            return $this->sendError($validator->errors(), 400);
+        }
+
+        $credentials = $request->only('email', 'password');
+        if (Auth::attempt($credentials)) {
+            $token = $request->user()->createToken('token');
+            return response()->json([
+                                        'token'      => $token->accessToken,
+                                        'expires_at' => $token->token->expires_at->toIso8601String()
+                                    ], 200)
+                             ->header('Authorization', $token->accessToken);
+        }
+        return response()->json(['error' => 'login_error'], 401);
+    }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     * @api v1
+     */
+    public function logout(Request $request):JsonResponse {
+        $isUser = $request->user()->token()->revoke();
+        if ($isUser) {
+            $success['message'] = "Successfully logged out.";
+            return $this->sendResponse($success);
```

Comment:
The way you create the array here is messy.

Please simply inline this as `['message' => 'Successfully logged out.']` or first define `$success` before $defining `$success['message']`.